sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -512,21 +512,16 @@ func cloudauthAccountFromResourceData(data *schema.ResourceData) *v2.CloudauthAc
 	featureData := data.Get(SchemaFeature)
 	accountFeatures := constructAccountFeatures(&cloudauth.AccountFeatures{}, featureData)
 
-	cloudAccount := &cloudauth.CloudAccount{
-		Enabled:        data.Get(SchemaEnabled).(bool),
-		OrganizationId: data.Get(SchemaOrganizationIDKey).(string),
-		ProviderId:     data.Get(SchemaCloudProviderId).(string),
-		Provider:       cloudauth.Provider(cloudauth.Provider_value[data.Get(SchemaCloudProviderType).(string)]),
-		Components:     accountComponents,
-		Feature:        accountFeatures,
-	}
-
-	if cloudAccount.Provider == cloudauth.Provider_PROVIDER_AZURE {
-		cloudAccount.ProviderTenantId = data.Get(SchemaCloudProviderTenantId).(string)
-	}
-
 	return &v2.CloudauthAccountSecure{
-		CloudAccount: *cloudAccount,
+		CloudAccount: cloudauth.CloudAccount{
+			Enabled:          data.Get(SchemaEnabled).(bool),
+			OrganizationId:   data.Get(SchemaOrganizationIDKey).(string),
+			ProviderId:       data.Get(SchemaCloudProviderId).(string),
+			Provider:         cloudauth.Provider(cloudauth.Provider_value[data.Get(SchemaCloudProviderType).(string)]),
+			Components:       accountComponents,
+			Feature:          accountFeatures,
+			ProviderTenantId: data.Get(SchemaCloudProviderTenantId).(string),
```

Comment:
Actually yes this makes sense. In case of other vendors, this will just be passed as empty. For Azure, if this is empty, BE will take care of returning a bad request for missing `provider_tenant_id`. And it will keep the provider lean/dumb.

Reply:
That's right! However, I'm conditionally adding this field in the `toResourceData`.